### ISSUE TITLE
StyleCop.Resharper + Code Cleanup + StyleCop rules

### ISSUE THREAD

**Issue by sguryev** [StyleCop/StyleCop.ReSharper#13]
Hi there.
I have a pretty simple task and last 4 hours I have been trying to implement it with almost no luck.
I hope you will be able to advise me and help.

**My config is:** 
VS2017 15.9.1
R# Ultimate 2018.2.3
StyleCop by JetBrains 2018.2.0

StyleCop settings:
![image](https://user-images.githubusercontent.com/6978458/48731115-522b3380-ec4d-11e8-85e6-66f7717ac407.png)
![image](https://user-images.githubusercontent.com/6978458/48731137-6111e600-ec4d-11e8-92e0-e0f02688ad31.png)

Code Cleanup profile:
![image](https://user-images.githubusercontent.com/6978458/48731161-76871000-ec4d-11e8-8315-ddb0eb12e840.png)

**My goals are:**
1. Generate the header for each file
2. Generate the doc stubs for each (public) member
3. Enable build errors for related (1, 2) StyleCop issues

**My steps (story):**
I have installed the StyleCop R# extension and restart the VS (btw, is there any notification that VS should be restarted? Did I miss it? I have spent about 20 mins on searching for the StyleCop after extension installing). I have got the `StyleCop` Code Cleanup profile and enabled the `Create XML doc stub` option. After the cleanup I was able to get the header and member docs. Unfortunately there was no company name int the header and it was not complained with one of the StyleCop rule. So I have started searching a way to resolve it. 

I have realized that there is no other way for comfortable editing the `Settings.StyleCop` file except installing the VS StyleCop extension (by right mouse button click on a project). I was able to update the company name and copyright text there. After running the code cleanup again I have lost the **summary** section of the header (which has been generated as copy of the doc for the class in the file). I have tried to play with the R# header settings but it looks like StyleCop wins over the R# for header config. Based on the pretty old blog post [here](https://blog.jetbrains.com/dotnet/2010/12/20/automating-file-header-information-with-resharper/) there was a setting:
![image](https://user-images.githubusercontent.com/6978458/48728564-53a52d80-ec46-11e8-990f-53ac9898ea17.png)
allowed to configure the header cleanup. But it has gone.

**My questions are:**
1. How to config the header in order to add the summary and copyright (both are generated) at the same time with Code Cleanup?
2. Should I install StyleCop as the separate VS extension or not in order to use the R# one?
3. How can I fix this error during opening the StyleCop settings (exception occurred while loading one of the StyleCop add-ins `TypeLoadException` from assembly `TestPlatform.ObjectModel`)
![image](https://user-images.githubusercontent.com/6978458/48728735-daf2a100-ec46-11e8-90dd-c826f15ed0f4.png). 
Right now I have to disable the VS StyleCop, restart the VS, enable the plugin and restart again. It works till the next VS restart.

I have noticed that Alt+Enter R# fix add the full header (copyright + summary):
![image](https://user-images.githubusercontent.com/6978458/48730861-9f5ad580-ec4c-11e8-879f-7dab6ea9f8e0.png)
![image](https://user-images.githubusercontent.com/6978458/48730922-c87b6600-ec4c-11e8-8282-7dbad1451113.png)
But the Code Cleanup one add a copyright only:
![image](https://user-images.githubusercontent.com/6978458/48731009-08424d80-ec4d-11e8-8d1d-dbca05c6945c.png)
**And Code Cleanup replaces the full header with the short one during the cleanup process.**

Thank you!

**Comment by sguryev** [comment]
UPD: After R# repairing I can't run the VS StyleCop even after reinstalling the extension

**Comment by sguryev** [comment]
```
2277 | ERROR | LegacySitePackage failed for package [StyleCop.VisualStudio.StyleCopVSPackage, StyleCop.VisualStudio, Version=5.0.0.0, Culture=neutral, PublicKeyToken=f904653c63bc2738]Source: 'mscorlib' Description: Could not load type 'Microsoft.VisualStudio.TestPlatform.ObjectModel.InProcDataCollector.InProcDataCollection' from assembly 'Microsoft.VisualStudio.TestPlatform.ObjectModel, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. System.TypeLoadException: Could not load type 'Microsoft.VisualStudio.TestPlatform.ObjectModel.InProcDataCollector.InProcDataCollection' from assembly 'Microsoft.VisualStudio.TestPlatform.ObjectModel, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.   at System.Reflection.RuntimeAssembly.GetExportedTypes(RuntimeAssembly assembly, ObjectHandleOnStack retTypes)   at System.Reflection.RuntimeAssembly.GetExportedTypes()   at StyleCop.StyleCopCore.LoadAddins(String path, Byte[] publicKey)   at StyleCop.StyleCopCore.Initialize(ICollection`1 addInPaths, Boolean loadFromDefaultPath)   at StyleCop.VisualStudio.StyleCopVSPackage.InitializeMenus()   at StyleCop.VisualStudio.StyleCopVSPackage.Initialize()   at Microsoft.VisualStudio.Shell.Package.Microsoft.VisualStudio.Shell.Interop.IVsPackage.SetSite(IServiceProvider sp) | {629EB7CC-69C2-43AC-9BC9-482B0F810C4E} | 80131522 | VisualStudio | 2018/11/19 20:35:07.976
-- | -- | -- | -- | -- | -- | --
2278 | ERROR | SetSite failed for package [StyleCop.VisualStudio.StyleCopVSPackage, StyleCop.VisualStudio, Version=5.0.0.0, Culture=neutral, PublicKeyToken=f904653c63bc2738](null) | {629EB7CC-69C2-43AC-9BC9-482B0F810C4E} | 80131522 | VisualStudio | 2018/11/19 20:35:07.979
2279 | ERROR | End package load [StyleCop.VisualStudio.StyleCopVSPackage, StyleCop.VisualStudio, Version=5.0.0.0, Culture=neutral, PublicKeyToken=f904653c63bc2738] | {629EB7CC-69C2-43AC-9BC9-482B0F810C4E} | 80131522 | VisualStudio | 2018/11/19 20:35:07.984
```

**Comment by sguryev** [comment]
Generated by Code Cleanup:
![image](https://user-images.githubusercontent.com/6978458/48735460-755be000-ec59-11e8-95b8-0f669e89265f.png)

Generated by Alt+Enter:
![image](https://user-images.githubusercontent.com/6978458/48735485-8c9acd80-ec59-11e8-9e66-ae7da5d3d245.png)
![image](https://user-images.githubusercontent.com/6978458/48735494-958b9f00-ec59-11e8-8963-60ed3fa98f86.png)

First one doesn't match the rule `SA1623 PropertySummaryDocumentationMustMatchAccessors`

**Comment by matkoch** [comment]
Hey Sergey, sorry for the delay. Could you clarify where these settings are coming from? https://user-images.githubusercontent.com/6978458/48728564-53a52d80-ec46-11e8-990f-53ac9898ea17.png
